File: KrishiSetu.Api/Models/WorkerProfile.cs
using System.ComponentModel.DataAnnotations;

namespace KrishiSetu.Api.Models
{
    public class WorkerProfile
    {
        [Key]
        public int Id { get; set; }
        
        public int WorkerId { get; set; }
        public User Worker { get; set; } = null!;
        
        public string Skills { get; set; } = string.Empty;
        public int ExperienceYears { get; set; }
        public decimal HourlyRate { get; set; }
        
        public string AvailabilityStatus { get; set; } = "Available";
        
        public string Bio { get; set; } = string.Empty;
        
        public DateTime? AvailableDate { get; set; }
        
        public bool IsApproved { get; set; } = true;

        public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    }
}

File: krishisetu-backend/src/main/java/com/krishisetu/backend/entity/WorkerProfile.java
package com.krishisetu.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Table(name = "worker_profiles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class WorkerProfile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnoreProperties("workerProfile")
    private User worker;

    private String skills = "";
    private int experienceYears = 0;
    private double hourlyRate = 0.0;
    private String availabilityStatus = "Available";
    private String bio = "";
    private LocalDate availableDate;
    private boolean isApproved = true;
}
